packages/dataclass-editor/src/DataClassEditor.tsx

```diff
@@ -1,18 +1,33 @@
-import { Flex, PanelMessage, Spinner } from '@axonivy/ui-components';
+import {
+  Flex,
+  PanelMessage,
+  ResizableHandle,
+  ResizablePanel,
+  ResizablePanelGroup,
+  SidebarHeader,
+  Spinner,
+  Toolbar,
+  ToolbarTitle
+} from '@axonivy/ui-components';
 import { IvyIcons } from '@axonivy/ui-icons';
 import { useQuery } from '@tanstack/react-query';
 import { useEffect, useMemo, useState } from 'react';
-import { DataClassMaster } from './components/dataclass/master/DataClassMaster';
+import { isEntityClass } from './components/dataclass/data/dataclass-utils';
+import { DataClassMasterContent } from './components/dataclass/master/DataClassMasterContent';
+import { DataClassMasterToolbarControls } from './components/dataclass/master/DataClassMasterToolbarControls';
 import './DataClassEditor.css';
 import { useClient } from './protocol/ClientContextProvider';
 import type { EditorProps } from './protocol/types';
 import { genQueryKey } from './query/query-client';
 
 function DataClassEditor(props: EditorProps) {
+  const [sidebar, setSidebar] = useState(true);
+
   const [context, setContext] = useState(props.context);
   useEffect(() => {
     setContext(props.context);
   }, [props]);
+  const [selectedDataClassFieldIndex, setSelectedDataClassFieldIndex] = useState<number>();
```

Comment:
Honestly I don't like such long names 🙈 Too hard to read for my eyes
Why not something simpler like: `selectedField`

All of here is currently DataClass so we don't need to mention this everwhere. And the Index comes a little bit from the type. (maybe a bad example, but you also don't call a name variable nameText to showcase that it is a string :) )